Suppress new gcc-15 -Wunterminated-string-initialization

denix0 opened the pull request:
GCC 15 now warns when character arrays are being initialized by strings
and terminating NUL character doesn't fit. GCC 15.1 also allows marking
such arrays with nonstring attribute to suppress the warning. W/o such
attribute, the warning becomes error due to the global -Werror. Add the
attribute accordingly.

mikkorapeli-linaro commented:
Fixes optee-test yocto builds from meta-arm.

jforissier commented:
Thanks @denix0

`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

jenswi-linaro commented:
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

rossburton commented:
FWIW, our clang builds fail with this patch:
```
./regression_4000_data.h:7917:78: error: unknown attribute 'nonstring' ignored [-Werror,-Wunknown-attributes]
```

mikkorapeli-linaro commented:
> FWIW, our clang builds fail with this patch:
> 
> ```
> ./regression_4000_data.h:7917:78: error: unknown attribute 'nonstring' ignored [-Werror,-Wunknown-attributes]
> ```

https://github.com/OP-TEE/optee_test/pull/797 should help by disabling Werror. Maybe we can backport this to meta-arm...

rossburton commented:
> #797 should help by disabling Werror. Maybe we can backport this to meta-arm...

Yes, I'd approve that.

denix0 commented:
Is clang behind gcc for once? :)

Anyway, according to the manual clang 21 does have support for this attribute, but in Yocto we have 20.1 for now:
https://clang.llvm.org/docs/AttributeReference.html#nonstring

mikkorapeli-linaro commented:
I hope #ifdef's for gcc/clang and/or versions are not needed... or feature checks?

denix0 commented:
This nonstring attribute has been added to clang only last week:
https://github.com/llvm/llvm-project/issues/137705

denix0 commented:
> > I hope #ifdef's for gcc/clang and/or versions are not needed... or feature checks?
> 
> `#if __has_attribute(nonstring)` should work

I've just updated the commit with this check.

Should I collect all the early reviews in the commit message, or wait for the new ones?

jforissier commented:
> > > I hope #ifdef's for gcc/clang and/or versions are not needed... or feature checks?
> > 
> > 
> > `#if __has_attribute(nonstring)` should work
> 
> I've just updated the commit with this check.
> 
> Should I collect all the early reviews in the commit message, or wait for the new ones?

`Acked-by`'s can usually be kept when changes are minor. `Reviewed-by`'s are a different story, but you have mine anyways:

`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

Thanks!

denix0 commented:
> > I've just updated the commit with this check.
> > Should I collect all the early reviews in the commit message, or wait for the new ones?
> 
> `Acked-by`'s can usually be kept when changes are minor. `Reviewed-by`'s are a different story, but you have mine anyways:
> 
> `Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`
> 
> Thanks!

Ok, updated the commit one more time with reviews and acks. Thanks for all the feedback.
